### ISSUE TITLE
Accessibility for selected elements should be using 'aria-selected'

### ISSUE THREAD

**Issue by dkasper-was-taken** [cerner/terra-core#1153]
# Issue Description
ButtonGroups, TableRows, ListItems, MenuItems should use the 'aria-selected' attribute for their selected entries.

## Issue Type
- [ ] New Feature
- [ ] Enhancement
- [ ] Bug
- [ ] Other

**Comment by jmsv6d** [comment]
Actually, aria-selected is not an appropriate attribute for buttons, axe accessibility tests failed on the  button group when we were using aria-selected and they were updated to use aria-pressed instead. Aria-selected is probably appropriate for the others.

**Comment by jmsv6d** [comment]
Note: TableRows are already using 'aria-selected'
